metadata/snippet/interface.go

```diff
@@ -25,3 +25,14 @@ package snippet
 func AppendComponent(component, snippet string, params map[string]string) (string, error) {
 	return appendComponent(component, snippet, params)
 }
+
+// SetComponentParams takes
+//
+//   component: the name of the new component to be modified.
+//   snippet: a jsonnet snippet resembling the current component parameters.
+//   params: the parameters to be set for 'component'.
+//
+// and returns the jsonnet snippet with the modified set of component parameters.
+func SetComponentParams(component, snippet string, params map[string]string) (string, error) {
```

Comment:
Not a huge deal, but historically we have used these public-and-private duplicate functions to have slightly different signatures for testing. If these are going to be the same signature, though, it seems like you might just make them one function.